frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -294,8 +294,15 @@ private void trackCurrentBranchChange(GitRepository repository) {
         if (unmanagedBranchNotification != null) {
           unmanagedBranchNotification.expire();
         }
-        if (gitMacheteRepositorySnapshot != null) {
-          val entry = gitMacheteRepositorySnapshot.getBranchLayout().getEntryByName(repositoryCurrentBranchName);
+        val snapshot = gitMacheteRepositorySnapshot;
+        val mainGitDirectory = GitVfsUtils.getMainGitDirectory(repository);
+        // 1. As for now, only the snapshot of the repository selected in Git Machete panel is available
+        // (not all snapshots of all repositories!).
+        // 2. The unmanaged branch notification works on the same snapshot as the one selected in Git Machete panel.
+        // Hence, we must assure that the current branch changed belongs to the same repository as the given snapshot.
+        // TODO(#1542): A handling of all repositories (not only selected) is a subject to of improvement.
```

Comment:
1. As for now, only the snapshot of the repository selected in Git Machete panel is available (not all snapshots of all repositories!).
2. The unmanaged branch notification works on the same snapshot as the one selected in Git Machete panel.

Hence, we must assure that the current branch changed belongs to the same repository as the given snapshot.

A handling of all repositories (not only selected) is a subject to of improvement.